src/controllers/settings.js

```diff
@@ -138,39 +132,51 @@ async function resendEmail (req, res) {
     throw fluentError('user-verify-token-error')
   }
 
-  await sendVerificationEmail(emailId)
+  await sendVerificationEmail(res, emailId)
+}
 
+async function sendVerificationEmail (res, emailId) {
+  try {
+    const unverifiedEmailAddressRecord = await resetUnverifiedEmailAddress(
+      emailId
+    )
+    const recipientEmail = unverifiedEmailAddressRecord.email
+    const data = {
+      recipientEmail,
+      ctaHref: getVerificationUrl(unverifiedEmailAddressRecord),
+      utmCampaign: 'email_verify',
+      unsubscribeUrl: getUnsubscribeUrl(
+        unverifiedEmailAddressRecord,
+        'account-verification-email'
+      ),
+      heading: getMessage('email-verify-heading'),
+      subheading: getMessage('email-verify-subhead'),
+      partial: { name: 'verify' }
+    }
+    await sendEmail(
+      recipientEmail,
+      getMessage('email-subject-verify'),
+      getTemplate(data, verifyPartial(data))
+    )
+  } catch (err) {
+    if (err.message === 'error-email-validation-pending') {
+      res.status(429)
```

Comment:
Here you can use `RateLimitError` class and it'll set the json automatically

Reply:
Oh this is great, thanks!